### PR TITLE
release: Remove all unsused gitlib functions

### PR DESCRIPTION
**Pull request by errordeveloper**
This code was copied from k/k and is certainly not in use, I checked all the references.

### CODE REVIEW

**Review by aanm — CHANGES_REQUESTED** [review]
actually I think the `contrib/release/relnotes` can be deleted as well. @joestringer do you use the `contrib/release/uplodarev`? I don't.

**Comment by errordeveloper** [comment]
@aanm @joestringer I would quite excited to add more cleanup commits here!

**Comment by aanm** [comment]
> actually I think the `contrib/release/relnotes` can be deleted as well. @joestringer do you use the `contrib/release/uplodarev`? I don't.

@errordeveloper given that Joe thumbs up I think you can delete those files as well

**Comment by errordeveloper** [comment]
@aanm thanks! I've update the PR and tried to delete all references to `relnotes` and `uploadarev`, however not sure what to do with [`Documentation/contributing/release/rc.rst`](https://github.com/cilium/cilium/blob/48f8e7996d7d3031410877326267b6650bf7fbb4/Documentation/contributing/release/rc.rst), it still mentions `uploadrev` and I doubt wether the whole file is up-to-date or not...

**Comment by aanm** [comment]
> @aanm thanks! I've update the PR and tried to delete all references to `relnotes` and `uploadarev`.
> 
> However, I'm not sure what to do with [`Documentation/contributing/release/rc.rst`](https://github.com/cilium/cilium/blob/48f8e7996d7d3031410877326267b6650bf7fbb4/Documentation/contributing/release/rc.rst), it still mentions `uploadrev` and I doubt wether the whole file is up-to-date or not...

@errordeveloper thanks. If you can remove the section as part of the PR:
```
Release binaries
----------------

<< contents of snippet outputed by uploadrev >>
```
that would be great as well.

**Review by aanm — APPROVED** [review]

File: contrib/release/lib/common.sh [hunk 1]
```diff
@@ -357,52 +357,6 @@ common::exit () {
   common::cleanexit $etype
 }
 
-#############################################################################
-# Simple yes/no prompt
-#
-# @optparam default -n(default)/-y/-e (default to n, y or make (e)xplicit)
-# @param message
-common::askyorn () {
```
Comment (joestringer) [inline contrib/release/lib/common.sh]: I was about to add a user for this, so we can either keep it in this PR or when I submit the user, issue a partial revert.

**Review by joestringer — APPROVED** [review]
Totally makes sense to get rid of scripts like `relnotes` / `uploadrev` that we don't use. :+1: 

I'm not sure I understand the motivation of gutting libraries that we pick up from other locations. If we ever update them, then we need to go through and figure out which parts we've edited and which parts we haven't, rather than just pulling them wholesale and updating.  That said, in this case I'm not sure we've ever updated the scripts and I'm not sure where they come from so I'm not really against removing functions we don't use.

There was one I was planning to use below, we can resolve this either now by not removing or later by re-adding the function when I actually introduce the script to use it. Completely reasonable request that I re-introduce it when I actually introduce a script to use it, given it's currently unused :-)

File: contrib/release/lib/common.sh (hunk 1 above)
Comment (errordeveloper) [reply contrib/release/lib/common.sh]: No worries, I've put this one back in!

**Comment by errordeveloper** [comment]
> I'm not sure where they come from

@joestringer these where from Kubernetes, and from what I can tell these had been rewritten, namely release note generator is now [a JavaScript app](https://github.com/kubernetes-sigs/release-notes), and [it has a UI](https://relnotes.k8s.io/).

**Pull request merged**